pkg/generate/app/sourcelookup.go

```diff
@@ -265,6 +269,26 @@ func (r *SourceRepository) AddDockerfile(contents string) error {
 	return nil
 }
 
+// AddBuildSecrets adds the defined secrets into a build. The input format for
+// the secrets is "<secretName>:<destinationDir>". The destinationDir is
+// optional and when not specified the default is the current working directory.
+func (r *SourceRepository) AddBuildSecrets(secrets []string) error {
```

Comment:
scratch that last comment ;-) I just tested:

```
oc new-build -D $'FROM centos:7\nRUN yum install -y httpd' --build-secret "foo:/var" -o json
```

And it works :-) (secrets are added properly).